### ISSUE TITLE
Escaping goes over the entire phrase instead of only the macro expansion

### ISSUE THREAD

**Issue by zimmerle** [SpiderLabs/ModSecurity#743]
Whenever a macro expansion is used it places a "marking" on the "rule" to escape the parameters of the @rx operator. The escaping goes over the entire parameter, making impossible to use a macro as part of a regex, as illustrated bellow: 

```
SecRule MATCHED_VARS "@rx [a]+%{TX.0}[b]+" "...."
```

The escaping should goes over the contents of the %{TX.0} only.

References:
https://github.com/SpiderLabs/ModSecurity/blob/master/apache2/re_operators.c#L461

Somehow related: #716, #462, #215.

**Comment by martinhsv** [comment]
I just saw this older issue, and suspect that it is out-of-date (or possibly misclassified).

In investigating #2357, I found that ModSecurity 3.0.4 does not escape special characters in either the macro value or in the static portion of the regex pattern.

In other words I don't believe this is an issue in ModSecurity 3.x.

It is a legitimate issue in v2.9.x, however.  Escaping the special characters in the static portion of the regex there restricts the flexibility of the rule writer and doesn't really add any value.

So, (unless I've misunderstood something) I think this could be closed as a duplicate of #2357 , which is specifically tagged with '2.x'
